File: week12/HelloWorld/HelloWorld.java
import java.io.*;
import akka.actor.*;

// -- MESSAGE --------------------------------------------------
class MyMessage implements Serializable { // must be Serializable:
	public final String s;
	public MyMessage(String s) { this.s = s; }
}

// -- ACTOR --------------------------------------------------
class MyActor extends UntypedActor {
	private int count = 0; // can have (local) state

	public void onReceive(Object o) throws Exception { // reacting to message:
		if (o instanceof MyMessage) {
			MyMessage message = (MyMessage) o;
			System.out.println(message.s + " (" + count + ")");
			count++;
		}
	}
}

public class HelloWorld {
	public static void main(String[] args) {
		final ActorSystem system = ActorSystem.create("HelloWorldSystem");
		final ActorRef myactor =
		system.actorOf(Props.create(MyActor.class), "myactor");
		
		myactor.tell(new MyMessage("hello"), ActorRef.noSender());
		myactor.tell(new MyMessage("world"), ActorRef.noSender());
		
		try {
			System.out.println("Press return to terminate...");
			System.in.read();
		} catch(IOException e) {
			e.printStackTrace();
		} finally {
			system.shutdown();
		}
	}
}

File: week12/HelloWorld/HelloWorld.erl
% hello world program: created here => http://www.tutorialspoint.com/compile_erlang_online.php
-module(helloworld).
-export([start/0, myactor/1]).

start() ->
    MyActor = spawn(helloworld, myactor, [0]),
    MyActor ! {msg, "hello"},
    MyActor ! {msg, "world"},
    MyActor ! {msg, "test"}.
    
myactor(Count) -> %% can have state, such as Counter here
    receive
        {msg, Msg} ->
            io:fwrite(Msg ++ " ("),
            io:write(Count),
            io:fwrite(")\n"),
            myactor(Count + 1)
    end.